R/apply_footnote_meta.R

```diff
@@ -0,0 +1,177 @@
+
+#' Locates the footnote for a given  footnote structure
+#'
+#' @param footnote_structure footnote structure
+#' @param .data data set
+#' @param col_plan_vars col plan variables
+#' @param element_row_grp_loc row group location element
+#' @param group group quosures
+#' @param label label quosure
+#' @param columns column quosure
+#'
+#' @return list with the row, column and footnote
+#' @noRd
+#' @importFrom purrr cross_df
+#' @importFrom dplyr inner_join first last cur_group_id
+locate_fn <- function(footnote_structure, .data, col_plan_vars, element_row_grp_loc,
+                            group, label, columns){
+  col_info <- get_col_loc(footnote_structure, .data, col_plan_vars, columns)
+
+  loc_info <- get_row_loc(footnote_structure, .data, element_row_grp_loc,
+                          group, label, col_info)
+  loc_info$note <- footnote_structure$footnote_text
+  loc_info
+}
+
+
+
+
+get_col_loc <- function(footnote_structure, .data, col_plan_vars, columns){
+  loc_info <- footnote_structure %>%
+    discard(is.null) %>%
+    .[names(.) != "footnote_text"]
+
+  # Get column information
+  if( "column_val" %in% names(loc_info)){
+    col_str <- columns %>% map_chr(as_label)
+
+    if(is_empty(names(loc_info$column_val))){
+      col_val_nm <- col_str
+      loc_col_df <- tibble(!!col_str := loc_info$column_val)
+    } else {
+      loc_col_df <- loc_info$column_val %>%
+        cross_df()
+      col_val_nm <- names(loc_info$column_val)
+    }
+
+    col_loc_df <- split_data_names_to_df(NULL, col_plan_vars%>%
+                                           map_chr(as_label), col_str) %>%
+      inner_join(loc_col_df, by = col_val_nm)
+
+    # if not a column return the spanning column name
+    span_lvl <- col_str %in% col_val_nm %>%
+      which() %>%
+      max() %>%
+      col_str[.]
+
+    if(last(col_str) == span_lvl){
+      col_loc <- unite_df_to_data_names(col_loc_df, preselected_cols = c(), column_names = col_str)
+      if(!is.null(names(col_loc))){
+        col_loc <- if_else(names(col_loc) != "", names(col_loc), col_loc)
+      }
+      out <- list(col = col_loc, spanning = FALSE)
+    } else {
+      col_loc <- col_loc_df %>%
+        pull(paste0("__tfrmt_new_name__", span_lvl)) %>%
+        unique()
+      out <- list(col = col_loc, spanning = TRUE)
+    }
+  } else {
+    out <- list(col = NULL, spanning = FALSE)
+  }
+  out
+}
+
+get_row_loc <- function(footnote_structure, .data, element_row_grp_loc,
+                        group, label, col_info){
+
+  loc_info <- footnote_structure %>%
+    discard(is.null) %>%
+    .[names(.) != "footnote_text"]
+
+  row_grp <- ifelse(is.null(element_row_grp_loc$location), "",
+                     element_row_grp_loc$location)
+   if(any(names(loc_info) %in% c("group_val", "label_val"))){
+     # Things that don't have rows
+     if(!is.null(col_info$spanning) && col_info$spanning){
+       warning("Can not apply footnotes to group cells when only the columns and rows are specified")
```

Comment:
Can you please explain the meaning of these warnings? I am confused about the language and intent here